src/Utilities/ContainerHelpers.hpp

```diff
@@ -122,3 +123,50 @@ SPECTRE_ALWAYS_INLINE decltype(auto) get_size(
       tt::is_complex_of_fundamental_v<std::remove_cv_t<T>> or
       cpp17::is_fundamental_v<std::remove_cv_t<T>>)>::get_size(t, size);
 }
+
+namespace ContainerHelpers_detail {
+template <typename T>
+auto get_used_for_size_impl(
+    const size_t size, T /*meta*/,
+    cpp17::bool_constant<false> /*is fundamental*/,
+    cpp17::bool_constant<true> /*is vectorimpl*/) noexcept {
+  return T(size);
+}
+
+template <typename T>
+auto get_used_for_size_impl(
+    const size_t /*size*/, T /*meta*/,
+    cpp17::bool_constant<true> /*is fundamental*/,
+    cpp17::bool_constant<false> /*is vectorimpl*/) noexcept {
+  return T{};
+}
+template <typename T, size_t S>
+auto get_used_for_size_impl(
+    const size_t size, std::array<T, S> /*meta*/,
+    cpp17::bool_constant<false> /*is fundamental*/,
+    cpp17::bool_constant<false> /*is vectorimpl*/) noexcept {
+  return get_used_for_size_impl(
+      size, T{}, cpp17::bool_constant<tt::is_complex_or_fundamental_v<T>>{},
+      cpp17::bool_constant<is_derived_of_vector_impl_v<T>>{});
+}
+}  // namespace ContainerHelpers_detail
+
+// @{
+/// \ingroup UtilitiesGroup
+/// Helper function for obtaining an appropriate object to pass to
+/// various utilities which take an object in order to use its size to create a
+/// similar sized object.
+template <typename T>
+auto get_used_for_size(size_t size) noexcept {
+  return ContainerHelpers_detail::get_used_for_size_impl(
+      size, T{}, cpp17::bool_constant<tt::is_complex_or_fundamental_v<T>>{},
+      cpp17::bool_constant<is_derived_of_vector_impl_v<T>>{});
+}
+
+template <typename T>
```

Comment:
wasn't there something like this already in `VectorImplTestHelper`?  If so, shouldn't it be removed?